zebra/zebra_rnh.c

```diff
@@ -354,42 +454,62 @@ zebra_rnh_resolve_import_entry(struct zebra_vrf *zvrf, afi_t afi,
 
 	if (re)
 		*prn = rn;
+
+	if (!re && IS_ZEBRA_DEBUG_NHT_DETAILED)
+		zlog_debug("\tRejected due to removed or is a bgp route");
+
 	return re;
 }
 
 /*
  * See if a tracked route entry for import (by BGP) has undergone any
  * change, and if so, notify the client.
  */
-static void zebra_rnh_eval_import_check_entry(vrf_id_t vrfid, afi_t afi,
+static void zebra_rnh_eval_import_check_entry(struct zebra_vrf *zvrf, afi_t afi,
```

Comment:
'vrfid' is replaced by 'zvrf' pointer, but only the vrf id value is used (in debugs) - was there a reason to hand the zvrf pointer in there?

Reply:
long term I'm passing around the zvrf I think further code changes will be allowed to simplify the code when we reduce nht changing down to one code path in the future